src/ghostfragment/fragmenting/inter_finder.hpp

```diff
@@ -0,0 +1,34 @@
+#include <chemist/molecule/fragmented_molecule.hpp>
+#include <chemist/nucleus/fragmented_nuclei.hpp>
+
+#include <algorithm>
+#include <cmath>
+#include <cstdlib>
```

Comment:
I don't think you need these three includes for declaring the functions. Since users of this file don't need these include files' contents, moving them to the source file helps avoid contaminating the namespaces of files which include `inter_finder.hpp`

Reply:
I see. I thought best practice would be to have all the includes in the header file to keep the source file as clean as possible. That is an easy fix.

Reply:
It's actually the opposite, *i.e.*, keep the header files as clean as possible. The header files define your code's API; in theory, that's the only part of your code that users care about. So having clean header files is more important.

Reply:
Oh, interesting. I will keep that in mind.